rust/kernel/platform.rs

```diff
@@ -180,6 +181,36 @@ impl Device {
         // SAFETY: By the type invariants, we know that `self.ptr` is non-null and valid.
         unsafe { (*self.ptr).id }
     }
+
+    /// Gets a system resources of a platform device.
+    pub fn get_resource(&self, rtype: IoResource, num: usize) -> Option<Resource> {
+        // SAFETY: `self.ptr` is valid by the type invariant.
+        let res = unsafe { bindings::platform_get_resource(self.ptr, rtype as _, num as _) };
+        if res.is_null() {
+            return None;
+        }
+
+        // SAFETY: The pointer `res` is returned from `bindings::platform_get_resource`
+        // above and checked if it is not a NULL.
+        unsafe { Resource::new((*res).start, (*res).end) }
+    }
+
+    /// Ioremaps resources of a platform device.
+    ///
+    /// # Safety
+    ///
+    /// Satisfy a safety requirements of the `IoMem<SIZE>::try_new` method.
```

Comment:
This appears to be an implementation detail. If we have the conditions set up in the next paragraph, why do we need this here?

Reply:
How I see it is that both `IoMem<SIZE>::try_new`'s (a) and (b) requirements may still be violated by the callers
so of the `get_resource()` method as they are still related to the returned interface. This looks like safety requirements
need to be inherited.

Reply:
I see, that's clear then.
Thanks!